### PR TITLE
ISSUE-1124: Enhancements to masking results of scans

### PR DESCRIPTION
**Pull request by varunsharma0286**
Fixes #
https://github.com/deepfence/ThreatMapper/issues/1124

Changes proposed in this pull request:
-
-
-

### CODE REVIEW

File: deepfence_worker/ingesters/cloud_compliance.go [hunk 1]
```diff
@@ -50,7 +52,27 @@ func CommitFuncCloudCompliance(ns string, data []ingestersUtil.CloudCompliance)
 func CloudCompliancesToMaps(ms []ingestersUtil.CloudCompliance) []map[string]interface{} {
 	res := []map[string]interface{}{}
 	for _, v := range ms {
-		res = append(res, v.ToMap())
+		data := v.ToMap()
+
+		sep := "--"
```
[the remark this answers is not in the document]
Comment (varunsharma0286) [reply deepfence_worker/ingesters/cloud_compliance.go]: Done.

File: deepfence_server/reporters/scan/scan_reporters.go [hunk 2]
```diff
@@ -697,7 +697,8 @@ func GetScanResultDiff[T any](ctx context.Context, scan_type utils.Neo4jScanType
 	MATCH (n:` + string(scan_type) + `{node_id: $base_scan_id}) -[r:DETECTED]-> (d)
 	WHERE NOT EXISTS {MATCH (m:` + string(scan_type) + `{node_id: $compare_to_scan_id}) -[:DETECTED]-> (d)}
 	OPTIONAL MATCH (d) -[:IS]-> (e)
-	WITH apoc.map.merge( e{.*}, d{.*, masked: coalesce(d.masked or r.masked, false), name: coalesce(e.name, d.name, '')}) AS d` +
+	WITH apoc.map.merge( e{.*}, d{.*, masked: coalesce(d.masked or r.masked or e.masked, false),
```
Comment (noboruma) [inline deepfence_server/reporters/scan/scan_reporters.go:700]: IIRC we said `r.masked` was not needed anymore - is it?

File: deepfence_worker/ingesters/common.go [hunk 3]
```diff
@@ -207,3 +207,39 @@ func anyCompleted(data []map[string]interface{}) bool {
...
+	query := `MATCH (s:` + scanType + `{node_id:'` + scanId + `'}) - [:SCANNED] -> (n)
+		WITH labels(n) as label, n
+		RETURN 
+		CASE 
+    		WHEN 'ContainerImage' IN label or 'Container' in label 
+			THEN [(ci:ContainerImage{node_id:n.docker_image_id}) - [:IS] -> (cis) | cis.node_id] 
+    		WHEN 'Node' IN label THEN [n.node_id]
+    		ELSE [""]
+		END
+		AS result`
```
Comment (noboruma) [inline deepfence_worker/ingesters/common.go:224]: `AS result` is superfluous

File: deepfence_worker/ingesters/secrets.go [hunk 4]
```diff
@@ -68,7 +73,24 @@ func secretsToMaps(data []ingestersUtil.Secret) []map[string]map[string]interfac
...
 		}
-		secret["node_id"] = utils.ScanIdReplacer.Replace(fmt.Sprintf("%v:%v", i.Rule.ID, i.Match.FullFilename))
+
+		entityId := ""
+		if _, ok := secret["scan_id"]; ok {
+			scanId := secret["scan_id"].(string)
+			var err error
+			entityId, err = getEntityIdFromScanID(scanId, string(utils.NEO4J_SECRET_SCAN), tx)
+			if err != nil {
+				//TODO::Should we treat this as hard error and return?
```
Comment (noboruma) [inline deepfence_worker/ingesters/secrets.go:83]: yes we should so the ingester can retry and no data is being lost

File: deepfence_worker/ingesters/common.go [hunk 5]
```diff
@@ -207,3 +210,18 @@ func anyCompleted(data []map[string]interface{}) bool {
 
 	return complete
 }
+
+func getEntityIdFromScanID(scanId string) string {
```
[the remark this answers is not in the document]
Comment (varunsharma0286) [reply deepfence_worker/ingesters/common.go]: Removed this as we have changed the logic for the `getEntityIdFromScanID`

File: deepfence_server/reporters/scan/scan_reporters.go (hunk 2 above)
Comment (varunsharma0286) [reply deepfence_server/reporters/scan/scan_reporters.go:700]: This is required for the Scan level masking.

File: deepfence_worker/ingesters/secrets.go (hunk 4 above)
Comment (varunsharma0286) [reply deepfence_worker/ingesters/secrets.go]: Done.

File: deepfence_worker/ingesters/common.go (hunk 3 above)
Comment (varunsharma0286) [reply deepfence_worker/ingesters/common.go]: Removed.

File: deepfence_worker/ingesters/common.go [hunk 6]
```diff
@@ -207,3 +207,38 @@ func anyCompleted(data []map[string]interface{}) bool {
...
+
+	entityId := ""
+	query := `MATCH (s:` + scanType + `{node_id:'` + scanId + `'}) - [:SCANNED] -> (n)
+		WITH labels(n) as label, n
+		RETURN 
+		CASE 
+    		WHEN 'ContainerImage' IN label or 'Container' in label 
+			THEN [(ci:ContainerImage{node_id:n.docker_image_id}) - [:IS] -> (cis) | cis.node_id] 
+    		WHEN 'Node' IN label THEN [n.node_id]
+    		ELSE [""]
```
Comment (noboruma) [inline deepfence_worker/ingesters/common.go:222]: Shall we merge those two together?
Default case we just return `n.node_id`

Comment (varunsharma0286) [reply deepfence_worker/ingesters/common.go]: Done.

**Review by noboruma — APPROVED** [review]

**Review by noboruma — APPROVED** [review]

**Pull request merged**